### DISCUSSION TITLE
Navigation button without route

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#64] [Q&A] [answered]
With the new "large" button designs previewed in the Navigation Options+Settings threads, it got me thinking about the main "Navigation" button and, when pressed, telling me *"There is no route"*.

Suggestion: In the "no route" scenario, a press would launch a *"Where to?"* box with "Home", "Favorite", "Load GPX" etc.

I'm not sure how this sits with the planned "offline" search - but it strikes me that in ANother Apps, the Navigate button *usually* offers you a recent/favourite destination when pressed.

**Comment by devemux86** [comment]
@MotoUKRider thanks for the suggestion.

Route planning is a prerequisite for navigation, so it should be done beforehand.

**Reply by 0709wiwiwi** [reply]
I still too sometimes click too quickly on the navigation button without loaded trajectory. I find the suggestion  "Where to" still valuable.  Offer as a first proposal  what you already had set before like "home". And given the optimalised search function direct to a poi or favourite is maybe a possibility ? Thanks to the optimised autorecalculation function you don't keep returning if you deviate for any reason from what Cruiser intially suggested.

**Comment by MotoUKRider** [comment]
How is the progress with this?  It seems a waste to have a on-screen button not do anything when Cruiser is "idle".  Currently, 4 button presses to get to the file list when loading a GPX route - I'm hoping that can be improved upon :)

I wonder if the "Import Route" dialog can be streamlined so that, rather than picking a file type, the process just looks at the file extension?
It's a fiddly box to encounter, especially in winter gloves!

**Reply by devemux86** [reply]
We could show the import route dialog if we press the navigation button without an existing route?

---

The import dialog always appears and the last used format is automatically selected.
So the only case is if we want to change the file format?

Unfortunately Android makes things complicated with its restrictions on external files.
I will see again what improvements are possible. :slightly_smiling_face:

**Reply by MotoUKRider** [reply]
Yes, the dialog box would be a sensible option - I had a feeling that "Android" would be a limitation somewhere here!

**Comment by 0709wiwiwi** [comment]
> We could show the import route dialog if we press the navigation button without an existing route?

Navigate from gps position to the "single point (set) Target" > Start.

**Comment by devemux86** [comment]
We need to not over complicate things?
The button clearly inform us that there is not a route available for navigation.

We can plan a new route, open a route from favorites, import an external route, etc.
There are too many choices.

How other navigators work in such case?

**Reply by MotoUKRider** [reply]
Loading a route/trip in CoPilot was a corn-maze of button presses, not much fun.

I originally wrote:
> Suggestion: In the "no route" scenario, a press would launch a "Where to?" box with "Home", "Favorite", "Load GPX" etc.

Perhaps we could build on this?  I wonder if there's interest for a quick-nav "Home" marker?  Perhaps 4 or 6 panes of actions that take us directly to the "thing" rather than have to dig down 3-4 levels?

Alternatively, could the navigation button, when idle, perform a custom action that we define in the Settings, ie: launch the Route Import, open the Favorite Route dialog etc?

**Reply by 0709wiwiwi** [reply]
No route loaded > tap navigation > Navigate to ?  "Set Destination"   (Image Locus)

![image](https://github.com/devemux86/cruiser/assets/54167201/2cfab9fc-6a8a-4977-aa3e-3c3027a71b87)

**Reply by devemux86** [reply]
> I wonder if there's interest for a quick-nav "Home" marker? Perhaps 4 or 6 panes of actions that take us directly to the "thing" rather than have to dig down 3-4 levels?

Do you mean make a menu (or other) button to perform a custom action?

I have thought about this before, better to discuss it in a new separate topic.

> Alternatively, could the navigation button, when idle, perform a custom action that we define in the Settings, ie: launch the Route Import, open the Favorite Route dialog etc?

Do you mean when there is no route?

If there is no route, we could show a menu:
- **Plan route**: shows the position buttons to set the destination
- **Load favorite route**: opens the favorite routes screen
- **Import route**: opens the import route screen
- ...

**Reply by MotoUKRider** [reply]
> Do you mean when there is no route?

Yes - it just seems a shame to waste an empty button when it could be utilized *somehow*

**Comment by devemux86** [comment]
We could show the menu that appears when long press on map.

Especially if we enhance it with more content, like the suggestion by @bios4 here:
- https://github.com/devemux86/cruiser/discussions/482#discussioncomment-9243984

**Comment by devemux86** [comment]
Please check the implementation in the latest:
- https://github.com/devemux86/cruiser/discussions/482

**Reply by bios4** [reply]
Importing a route works properly per pressing the navigation button.

What is somewhat "disturbing" is the toast message telling "There is no route" which is displayed over the now new menu when pressing the button:

<img src="https://github.com/devemux86/cruiser/assets/145460147/40c25276-b2b3-4822-85e1-33da401ec4f8" width="33%">

Also the long-press on map menu looks nice. 👍

**Reply by devemux86** [reply]
> the toast message telling "There is no route" which is displayed over the now new menu when long-pressing the button

Somehow informing the user about the absence of a route,
and why they see this dialog and not the expected navigation.

**Reply by bios4** [reply]
Then maybe change the text to "There is no route _loaded yet_"?

**Reply by devemux86** [reply]
What is the difference with existing text?

The problem with message variations is that we have to translate (and maintain) them in (currently) 11 languages.
Current text is also used in many other places.

**Reply by bios4** [reply]
> What is the difference with existing text?

If I open the menu with the attention to either create a new route or import one (when I use this menu I am perfectly aware of the fact that no route is loaded), then it could be irritating that the app tells me "there is no route"...

One might think at first "What, there is no route to import?" 😮
Or "Why is there no route I can create?"

But for sure this is just a minor glitch.

**Reply by devemux86** [reply]
The message only appears when press the navigation button (not when long press the map).

Pressing the navigation button has primary purpose the navigation.
Displaying a menu may confuse (especially new) users with the menu button?

That's why I kept the message and just added the menu. Don't know if it is the best workflow.

**Comment by devemux86** [comment, marked as answer]
Implemented in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by ramondevic** [comment]
This makes it easier to access the final destination, a waypoint, import a route, etc. instead of giving a message that there is no route defined. Great !

**Comment by MotoUKRider** [comment]
This rather useful (possibly more so to newbies) feature seems to have been removed at some point.
eg: Navigation button - Import (straight to file picker)

There is a spare "hole" in the tile menu 👽

**Reply by devemux86** [reply]
We could add it.

**Reply by MotoUKRider** [reply]
Yes - I'm thinking the Navigate (or "Where to?") button is a good place to include it as we have that spare Tile as well as the bigger menu that offers Start/End/Extend etc.

**Reply by devemux86** [reply]
It does not seem to belong there, as users have not pressed navigation yet.
Or they may just want to use the follow location function.

It belongs more in the menu that opens after pressing navigation without a route,
but a more simplified menu:

<img src="https://github.com/user-attachments/assets/65a0d35e-1879-4b28-a0c6-512b6716a54a" width="50%"/>

**Reply by MotoUKRider** [reply]
@devemux86 That's an ideal place to put it!
